salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+            description {[type]} -- A human-friendly description
+        """
+
+        config = config or {}
+
+        self.type = type
+        self.path = path.replace("/", "")
+        self.description = (description if description else "")
+        self.config = {k: v for k, v in config.items() if v != ''}
+
+    def get_unique_id(self):
+        """Return a unique hash of the config by only using the type and path
+
+        Returns:
+            string -- unique hash of the type and path
+        """
+        return self.hash_value(self.type + self.path)
+
+    def get_tuning_hash(self):
+        """Return a unique ID per tuning configuration
+
+        Returns:
+            string -- unique hash of the configuration
+        """
+        return self.hash_value(self.description + str(self.config))
+
+    def hash_value(self, value):
+        return hashlib.sha256(value.encode()).hexdigest()
+
+    def __eq__(self, other):
+        return self.get_unique_id() == other.get_unique_id()
+
+    def __repr__(self):
+        return ("Path: %s - Type: %s - Desc: %s - Config: %s - Hash : %s" %
+                (self.path, self.type, self.description, str(self.config), self.get_unique_id()))
+
+
+class VaultAuthMethod(VaultConfigBase):
+    auth_config = None
+    extra_config = None
+
+    def __init__(self, type, path, description, config=None, auth_config=None, extra_config=None):
+        super().__init__(type, path, description, config)
+
+        self.auth_config = auth_config or {}
+        self.extra_config = extra_config or {}
+
+
+class VaultSecretEngine(VaultConfigBase):
+    secret_config = None
+    extra_config = None
+
+    def __init__(self, type, path, description, config=None, secret_config=None, extra_config=None):
+        super().__init__(type, path, description, config)
+
+        self.secret_config = secret_config or {}
+        self.extra_config = extra_config or {}
+
+
+class VaultAuditDevice(VaultConfigBase):
+
+    def __init__(self, type, path, description, config=None):
+        super().__init__(type, path, description, config)
+
+
+class VaultPolicyManager():
+    """Module for handling Vault Policies
+    """
+
+    def __init__(self):
+        """Initialize Vault Policies Manager
+        """
+        log.info("Initializing Vault Policies Manager...")
+
+    def get_remote_policies(self, client, ret):
+        """Retrieve policies from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            [list] -- policies
+        """
+        log.info('Retrieving policies from vault...')
+        polices = []
+        try:
+            policies_resp = client.sys.list_policies()
+
+            for policy in policies_resp['data']['policies']:
+                if not (policy == 'root' or policy == 'default'):
+                    polices.append(policy)
+
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.